bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVHandler.java

```diff
@@ -0,0 +1,575 @@
+                if (thingConfig.getUDN().isEmpty()) {
+                    // get UDN from device name
+                    String uid = this.getThing().getUID().getAsString();
+                    thingConfig.setUDN(StringUtils.substringAfterLast(uid, ":"));
+                }
+                if (thingConfig.getMacAddress().isEmpty()) {
+                    // get MAC address from UDN (last 12 digits)
+                    String macAddress = StringUtils.substringAfterLast(thingConfig.getUDN(), "_");
+                    if (macAddress.isEmpty()) {
+                        macAddress = StringUtils.substringAfterLast(thingConfig.getUDN(), "-");
+                    }
+                    thingConfig.setMacAddress(macAddress);
+                }
+
+                control = new MagentaTVControl(thingConfig, network);
+                thingConfig.updateConfig(control.getConfig().getProperties()); // get network parameters from control
+                authenticateUser();
+
+                // wait for NotifyServlet to initialze
+                if (!handlerFactory.getNotifyServletStatus()) {
+                    logger.debug("Waiting on NotifyServlet to start...");
+                    int iRetries = 30;
+                    while ((iRetries-- > 0) && !handlerFactory.getNotifyServletStatus()) {
+                        logger.trace("Waiting for init, {0} sec remaining", iRetries);
+                        Thread.sleep(1000);
+                    }
+                    if ((iRetries <= 0) && !handlerFactory.getNotifyServletStatus()) {
+                        throw new MagentaTVException("Can't initialize, NotifyServlet not started!");
+                    }
+                }
+
+                connectReceiver(); // throws MagentaTVException on error
+                // change to ThingStatus.ONLINE will be done when the pairing result is received
+                // (see onPairingResult())
+            } catch (MagentaTVException | InterruptedException | RuntimeException e) {
+                errorMessage = e.getMessage() != null ? e.getMessage() : e.getClass().toString();
+            } finally {
+                if (!errorMessage.isEmpty()) {
+                    setOnlineState(ThingStatus.OFFLINE, errorMessage);
+                }
+            }
+        });
+    }
+
+    /**
+     * Handle channel commands
+     *
+     * @param channelUID - the channel, which received the command
+     * @param command - the actual command (could be instance of StringType,
+     *            DecimalType or OnOffType)
+     */
+    @SuppressWarnings("null")
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (command == RefreshType.REFRESH) {
+            // currently no channels to be refreshed
+            return;
+        }
+
+        try {
+            if (!isOnline() || command.toString().equalsIgnoreCase("PAIR")) {
+                logger.info("Device {0} is offline, try to (re-)connect", deviceName());
```

Comment:
Change to logger.debug

Reply:
from my point of view this is an important information. how should the user know that the device is offline, ok we could check the thing status, but that means looking at 2 locations

Reply:
But that‘s the usual way.

Reply:
this is the recovery function to re-connect the device (geeting out from offline state)